drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h

```diff
@@ -0,0 +1,1099 @@
+ *	    supplied while invoking further device specific API's,
+ *	    e.g. wifi_nrf_fmac_scan() etc.
+ *
+ * @return Pointer to the context of the RPU instance.
+ */
+struct wifi_nrf_fmac_dev_ctx *wifi_nrf_fmac_dev_add(struct wifi_nrf_fmac_priv *fpriv,
+						    void *os_dev_ctx);
+
+
+/**
+ * @brief Removes a RPU instance.
+ * @param fmac_dev_ctx Pointer to the context of the RPU instance to be removed.
+ *
+ * @details This function handles the removal of an RPU instance at the UMAC IF layer.
+ *
+ * @return None.
+ */
+void wifi_nrf_fmac_dev_rem(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx);
+
+
+/**
+ * @brief Initializes a RPU instance.
+ * @param fmac_dev_ctx Pointer to the context of the RPU instance to be removed.
+ * @param rf_params_usr RF parameters (if any) to be passed to the RPU.
+ * @param sleep_type Type of RPU sleep.
+ * @param phy_calib PHY calibration flags to be passed to the RPU.
+ * @param op_band Operating band of the RPU.
+ * @param tx_pwr_ctrl_params TX power control parameters to be passed to the RPU.
+ *
+ * @details This function initializes the firmware of an RPU instance.
+ *
+ * @return Command execution status
+ */
+enum wifi_nrf_status wifi_nrf_fmac_dev_init(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
+			unsigned char *rf_params_usr,
+#ifdef CONFIG_NRF_WIFI_LOW_POWER
+			int sleep_type,
+#endif /* CONFIG_NRF_WIFI_LOW_POWER */
+			unsigned int phy_calib,
+			unsigned int op_band,
+			struct nrf_wifi_umac_tx_pwr_ctrl_params *tx_pwr_ctrl_params);
+
+
+/**
+ * @brief De-initializes a RPU instance.
+ * @param fmac_dev_ctx Pointer to the context of the RPU instance to be removed.
+ *
+ * @details This function de-initializes the firmware of an RPU instance.
+ *
+ * @return None.
+ */
+void wifi_nrf_fmac_dev_deinit(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx);
+
+
+/**
+ * @brief Loads the Firmware(s) to the RPU WLAN device.
+ * @param fmac_dev_ctx Pointer to the UMAC IF context for a RPU WLAN device,
+ *      which was passed as \p fmac_dev_ctx parameter via the
+ *      \p add_dev_callbk_fn() callback function.
+ * @param fmac_fw Information about the FullMAC firmware(s) to be loaded.
+ *
+ * @details This function loads the FullMAC firmware(s) to the RPU WLAN device.
+ *
+ * @return Command execution status
+ */
+enum wifi_nrf_status wifi_nrf_fmac_fw_load(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
+					   struct wifi_nrf_fmac_fw_info *fmac_fw);
+
+
+/**
+ * @brief Get FW versions from the RPU.
+ * @param fmac_dev_ctx Pointer to the UMAC IF context for a RPU WLAN device.
+ * @param fw_ver Pointer to the address where the FW versions needs to be copied.
+ *
+ * @details This function is used to get Firmware versions from the RPU.
+ *
+ * @return Command execution status
+ */
+enum wifi_nrf_status wifi_nrf_fmac_ver_get(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
+					  unsigned int *fw_ver);
+
+
+/**
+ * @brief Configure BT-Coex parameters in RPU.
```

Comment:
Can we write "Bluetooth coexistence" instead of "BT-Coex"?
```suggestion
 * @brief Configure BT-Coex parameters in RPU.
```